src/main/java/com/_up/megastore/services/interfaces/IProductService.java

```diff
@@ -14,7 +14,8 @@ public interface IProductService {
 
     Product findProductByIdOrThrowException(UUID productId);
 
-    void ifProductIsNotDeletedThrowException(UUID productId);
+    ProductResponse deleteProduct(UUID productId);
```

Comment:
Con las correcciones anteriores esto quedaría así
```suggestion
    void deleteProduct(UUID productId);
```